source/vulkan/vulkan_hooks.cpp

```diff
@@ -454,14 +688,449 @@ VkResult VKAPI_CALL vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR *pPr
+		if (clearValue.depthStencil.depth > 0)
+			depthstencilCleared = true;
+	}
+
+	if(depthstencilCleared)
+		{ const std::lock_guard<std::mutex> lock(s_mutex);
+			s_framebuffer_per_command_buffer[commandBuffer] = pRenderPassBegin->framebuffer;
+			const auto it = s_depth_stencil_buffer_frameBuffers.find(pRenderPassBegin->framebuffer);
+			if (it != s_depth_stencil_buffer_frameBuffers.end())
+			{
+				// retrieve the current depth stencil image view
+				reshade::vulkan::draw_call_tracker *command_buffer_tracker = &s_trackers_per_command_buffer.at(commandBuffer);
+				image_view_data image_view_data = it->second.image_view_data;
+				VkImageView depthstencil = it->second.imageView;
+				if (image_view_data.image_data.image_info.usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)
+					track_cleared_depthstencil(commandBuffer, device, *command_buffer_tracker, depthstencil, image_view_data);
+			}
+		}
+
+	trampoline(commandBuffer, pRenderPassBegin, contents);
+}
+
+void vkCmdDraw(VkCommandBuffer commandBuffer, uint32_t vertexCount, uint32_t instanceCount, uint32_t firstVertex, uint32_t firstInstance)
+{
+	PFN_vkCmdDraw trampoline = nullptr;
+
+	// no device associated (this cannot happen normally)
+	if (const auto it = s_command_buffer_mapping.find(commandBuffer); it == s_command_buffer_mapping.end())
+		return;
+
+	VkDevice device = s_command_buffer_mapping.at(commandBuffer);
+
+	trampoline = s_device_dispatch.at(get_dispatch_key(device)).CmdDraw;
+	assert(trampoline != nullptr);
+
+	trampoline(commandBuffer, vertexCount, instanceCount, firstVertex, firstInstance);
+
+	// retrieve the current depth stencil image view
+	reshade::vulkan::draw_call_tracker *command_buffer_tracker = &s_trackers_per_command_buffer.at(commandBuffer);
+	command_buffer_tracker->on_draw(command_buffer_tracker->_depthstencil, vertexCount * instanceCount);
+}
+
+void vkCmdDrawIndexed(VkCommandBuffer commandBuffer, uint32_t indexCount, uint32_t instanceCount, uint32_t firstIndex, int32_t vertexOffset, uint32_t firstInstance)
+{
+	PFN_vkCmdDrawIndexed trampoline = nullptr;
+
+	// no device associated (this cannot happen normally)
+	if (const auto it = s_command_buffer_mapping.find(commandBuffer); it == s_command_buffer_mapping.end())
+		return;
+
+	VkDevice device = s_command_buffer_mapping.at(commandBuffer);
+
+	trampoline = s_device_dispatch.at(get_dispatch_key(device)).CmdDrawIndexed;
+	assert(trampoline != nullptr);
+
+	trampoline(commandBuffer, indexCount, instanceCount, firstIndex, vertexOffset, firstInstance);
+
+	// retrieve the current depth stencil image view
+	reshade::vulkan::draw_call_tracker *command_buffer_tracker = &s_trackers_per_command_buffer.at(commandBuffer);
+	command_buffer_tracker->on_draw(command_buffer_tracker->_depthstencil, indexCount * instanceCount);
+}
+
+void vkCmdDrawIndexedIndirect(VkCommandBuffer commandBuffer, VkBuffer buffer, VkDeviceSize offset, uint32_t drawCount, uint32_t stride)
```

Comment:
No need to hook this. It is impossible to get draw call statistics from an indirect draw call, so I'd just ignore those.

Reply:
OK. Removed...